MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java

```diff
@@ -0,0 +1,291 @@
+     * @param randomDeath if this is for random divorce or manual divorce
+     * @return null if they can, otherwise the reason they cannot
+     */
+    public @Nullable String canDie(final Person person, final AgeGroup ageGroup,
+                                   final boolean randomDeath) {
+        if (person.getStatus().isDead()) {
+            return resources.getString("cannotDie.Dead.text");
+        } else if (randomDeath) {
+            if (person.isImmortal()) {
+                return resources.getString("cannotDie.Immortal.text");
+            } else if (!getEnabledAgeGroups().get(ageGroup)) {
+                return resources.getString("cannotDie.AgeGroupDisabled.text");
+            } else if (!isUseRandomClanPersonnelDeath() && person.isClanner()) {
+                return resources.getString("cannotDie.RandomClanPersonnel.text");
+            } else if (!isUseRandomPrisonerDeath() && person.getPrisonerStatus().isPrisoner()) {
+                return resources.getString("cannotDie.RandomPrisoner.text");
+            }
+        }
+
+        return null;
+    }
+
+    //region New Day
+    /**
+     * Processes new day random death for an individual.
+     * @param campaign the campaign to process
+     * @param today the current day
+     * @param person the person to process
+     */
+    public boolean processNewDay(final Campaign campaign, final LocalDate today, final Person person) {
+        final int age = person.getAge(today);
+        final AgeGroup ageGroup = AgeGroup.determineAgeGroup(age);
+        if (canDie(person, ageGroup, true) != null) {
+            return false;
+        }
+
+        if (randomlyDies(age, person.getGender())) {
+            person.changeStatus(campaign, today, getCause(person, ageGroup, age));
+            return person.getStatus().isDead();
+        } else {
+            return false;
+        }
+    }
+
+    //region Random Death
+    /**
+     * @param age the person's age
+     * @param gender the person's gender
+     * @return true if the person is selected to randomly die, otherwise false
+     */
+    public abstract boolean randomlyDies(final int age, final Gender gender);
+    //endregion Random Death
+    //endregion New Day
+
+    //region Cause
+    /**
+     * @param person the person who has died
+     * @param ageGroup the person's age group
+     * @param age the person's age
+     * @return the cause of the Person's random death
+     */
+    public PersonnelStatus getCause(final Person person, final AgeGroup ageGroup, final int age) {
+        if (person.getStatus().isMIA()) {
+            return PersonnelStatus.KIA;
+        } else if (person.hasInjuries(false)) {
+            final PersonnelStatus status = determineIfInjuriesCausedTheDeath(person);
+            if (!status.isActive()) {
+                return status;
+            }
+        }
+
+        if (person.isPregnant()) {
+            return PersonnelStatus.PREGNANCY_COMPLICATIONS;
```

Comment:
Is this really always the case?

Reply:
It is the design for now.

This should be a rare enough condition that this simplification should be fine, and I will be adjusting the cause generation with user feedback involved.